Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-20.ca-central-1.compute.internal
 Git:  git version 2.9.5
Sha: 94c37cd76aa34828f752f4641a1941f2d8824f5c
Number of Error(s): 4
src-react/components/FileDetails.tsx(161,18): error TS2304: Cannot find name isDFUWorkunit.
make[2]: *** [esp/src/CMakeFiles/eclwatch.dir/build.make:312: esp/src/build/dist/stub.eclwatch.js] Error 1
make[1]: *** [CMakeFiles/Makefile2:10418: esp/src/CMakeFiles/eclwatch.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-06-18_13-58-24
time logs started
Prep time   :     8 sec (00:00:08)
Build time  :   164 sec (00:02:44)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   172 sec (00:02:52)
time logs end

[Build error:](https://gist.github.com/2ef68c0dced7044d9630a56c7b5c71ce)
Build log [Build log](https://gist.github.com/HPCCSmoketest/2ef68c0dced7044d9630a56c7b5c71ce#file-result-21-06-18-13-55-18-log)

--- a/esp/src/src-react/components/FileDetails.tsx
+++ b/esp/src/src-react/components/FileDetails.tsx
@@ -6,10 +6,12 @@ import * as Utility from "src/Utility";
 import { getStateImageName, IFile } from "src/ESPLogicalFile";
 import { useFile } from "../hooks/File";
 import { pivotItemStyle } from "../layouts/pivot";
+import { DojoAdapter } from "../layouts/DojoAdapter";
 import { pushUrl } from "../util/history";
 import { ShortVerticalDivider } from "./Common";
 import { TableGroup } from "./forms/Groups";
 import { Queries } from "./Queries";
+import { WorkunitDetails } from "./WorkunitDetails";
 
 import "react-reflex/styles.css";
 
@@ -30,6 +32,7 @@ export const FileDetails: React.FunctionComponent<FileDetailsProps> = ({
     const [_protected, setProtected] = React.useState(false);
     const [restricted, setRestricted] = React.useState(false);
 
+    const isDFUWorkunit = file?.Wuid[0] === "D";
     const isProtected = file?.ProtectList?.DFUFileProtect?.length > 0 || false;
 
     React.useEffect(() => {
@@ -160,6 +163,7 @@ export const FileDetails: React.FunctionComponent<FileDetailsProps> = ({
             <PivotItem headerText={nlsHPCC.Graphs} itemKey="Graphs" style={pivotItemStyle(size, 0)}>
             </PivotItem>
             <PivotItem headerText={nlsHPCC.Workunit} itemKey="Workunit" style={pivotItemStyle(size, 0)}>
+                {isDFUWorkunit ? <DojoAdapter widgetClassID="DFUWUDetailsWidget" params={{ Wuid: file?.Wuid }} /> : <WorkunitDetails wuid={file?.Wuid} />}
             </PivotItem>
             <PivotItem headerText={nlsHPCC.History} itemKey="History" style={pivotItemStyle(size, 0)}>
             </PivotItem>

